### PR TITLE
Update spine-csharp to v3.2 (shearing)

### PR DESCRIPTION
**Pull request by pharan**

### CODE REVIEW

**Comment by NathanSweet** [comment]
@pharan can you add latest changes from spine-libgdx?

**Comment by pharan** [comment]
aw. sucks. waffle doesn't show updates to PRs. (already did this)

**Comment by endlesstravel** [comment]
Glad to know the good news, but I must say All the runtime library update to 3.0+ Except lua and lua-relevant runtime library :broken_heart:  . And I am anxious it will come. :joy:

File: spine-csharp/src/Animation.cs [hunk 1]
```diff
@@ -249,13 +249,13 @@ public RotateTimeline (int frameCount)
 			}
 
 			// Interpolate between the previous frame and the current frame.
-			int frameIndex = Animation.binarySearch(frames, time, 2);
-			float prevFrameValue = frames[frameIndex - 1];
-			float frameTime = frames[frameIndex];
-			float percent = 1 - (time - frameTime) / (frames[frameIndex + PREV_FRAME_TIME] - frameTime);
-			percent = GetCurvePercent((frameIndex >> 1) - 1, percent < 0 ? 0 : (percent > 1 ? 1 : percent));
+			int frame = Animation.binarySearch(frames, time, 2);
```
Comment (badlogic) [inline spine-csharp/src/Animation.cs]: Why the renaming?

File: spine-csharp/src/SkeletonBinary.cs [hunk 2]
```diff
@@ -40,11 +44,12 @@
 
 namespace Spine {
 	public class SkeletonBinary {
-		public const int TIMELINE_SCALE = 0;
-		public const int TIMELINE_ROTATE = 1;
-		public const int TIMELINE_TRANSLATE = 2;
-		public const int TIMELINE_ATTACHMENT = 3;
-		public const int TIMELINE_COLOR = 4;
+		public const int TIMELINE_ROTATE = 0;
```
Comment (badlogic) [inline spine-csharp/src/SkeletonBinary.cs]: Has the order of these things changed?

File: spine-csharp/src/Animation.cs (hunk 1 above)
Comment (pharan) [inline spine-csharp/src/Animation.cs]: Ask Nate. I'm just copying from libGDX.

File: spine-csharp/src/SkeletonBinary.cs (hunk 2 above)
Comment (pharan) [inline spine-csharp/src/SkeletonBinary.cs]: Yes, Nate changed the order in 3.2. They should match libGDX.

Comment (NathanSweet) [inline spine-csharp/src/SkeletonBinary.cs]: Any change (even just adding a value) to the binary format means previously exported data needs to be exported again, so changing the constants doesn't hurt. The change is to keep a consistent order for transform properties, always: rotate, translate, scale, then shear.

File: spine-csharp/src/Animation.cs (hunk 1 above)
Comment (NathanSweet) [inline spine-csharp/src/Animation.cs]: When adding TransformConstraintTimeline and I had a `FRAME_PREV_TRANSLATE_MIX`, I decided the `FRAME_` prefix wasn't helpful and made everything super long. I updated all the other timelines to keep things consistent. Spring cleaning!

**Comment by Tranmet** [comment]
Hi! How do I update spine runtime 3.1.8 spine-csharp and spine-unity to this 3.2? Is there any way I can pull the files or do I have to manually copy the files?

**Comment by NathanSweet** [comment]
Until this PR is merged, you can go to pharan's spine-runtimes repo, choose [branch csharp-3-2beta](https://github.com/pharan/spine-runtimes/tree/csharp-3-2beta), then click Clone or download and choose [zip](https://github.com/pharan/spine-runtimes/archive/csharp-3-2beta.zip).

**Comment by Tranmet** [comment]
Thanks alot! :)

**Comment by pharan** [comment]
@Tranmet you can also find a link to a unitypackaged version of spine-csharp 3.2 in this thread: http://esotericsoftware.com/forum/Noteworthy-Spine-Unity-Topics-5924

**Comment by pharan** [comment]
Just a note. 3.2 is both Shearing and keyable TransformConstraint.

**Comment by NathanSweet** [comment]
Forgot monogame and XNA readmes. :)

**Pull request merged**